src/control/server/ctl_storage.go

```diff
@@ -58,16 +59,18 @@ func NewStorageControlService(log logging.Logger, bdev *bdev.Provider, scm *scm.
 	}
 }
 
-func (c *StorageControlService) findBdevsWithDomain(sr *bdev.ScanResponse, prefix string) ([]string, error) {
+// findBdevsWithDomain retrieves controllers in scan response that match the
+// input prefix in the domain component of their PCI address.
+func findBdevsWithDomain(scanResp bdev.ScanResponse, prefix string) ([]string, error) {
```

Comment:
Out of curiosity, why did you switch these functions from `*bdev.ScanResponse` to `bdev.ScanResponse`?

Reply:
once we receive the response we can dereference to avoid having to check for a nil reference in each of the functions the response is passed to

Reply:
I think for unexported functions it's fine to avoid the nil check as the expectation is that any public methods will have already done the check before using the internal helpers. My concern with this approach is that there will be a lot of unnecessary copying, particularly with large/complex ScanResponse structs. I doubt that this will be significant overhead in practice, but I think as a general rule it's preferable to pass by reference rather than value.

Reply:
fixed